### PR TITLE
Trust Region docstrings improvement

### PR DESCRIPTION
**Pull request by yash2798**

### CODE REVIEW

**Comment by yash2798** [comment]
@ChrisRackauckas  this can be merged

File: src/trustRegion.jl [hunk 1]
```diff
@@ -88,6 +85,14 @@ for large-scale and numerically-difficult nonlinear systems.
     Currently, the linear solver and chunk size choice only applies to in-place defined
     `NonlinearProblem`s. That is expected to change in the future.
 """
+EnumX.@enumx RadiusUpdateSchemes begin
```
Comment (ChrisRackauckas) [inline src/trustRegion.jl:88]: why was this moved? This puts the docstrong on the wrong object.

Comment (yash2798) [reply src/trustRegion.jl:88]: Umm right now I dont see the trust region docstring on the main docs: https://docs.sciml.ai/NonlinearSolve/stable/api/nonlinearsolve/ . I thought it is because the docstring is not at the beginning.

Comment (yash2798) [reply src/trustRegion.jl:88]: So should the Enum definition be on the main `src/NonlinearSolve.jl` file instead?

Comment (ChrisRackauckas) [reply src/trustRegion.jl:88]: It's because the docstring wasn't right before the function. It has to be exactly the line before the function definition, no space.

Comment (ChrisRackauckas) [reply src/trustRegion.jl]: It's already working on the dev version: https://docs.sciml.ai/NonlinearSolve/dev/api/nonlinearsolve/
